static/app/components/events/groupingInfo/index.tsx

```diff
@@ -22,116 +23,66 @@ const groupingFeedbackTypes = [
   t('Other grouping issue'),
 ];
 
-type Props = DeprecatedAsyncComponent['props'] & {
+type GroupingInfoProps = {
   event: Event;
-  organization: Organization;
   projectSlug: string;
   showGroupingConfig: boolean;
   group?: Group;
 };
 
-type State = DeprecatedAsyncComponent['state'] & {
-  configOverride: string | null;
-  groupInfo: EventGroupInfo;
-  isOpen: boolean;
-};
-
-class GroupingInfo extends DeprecatedAsyncComponent<Props, State> {
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const {organization, event, projectSlug, group} = this.props;
-
-    if (
-      event.occurrence &&
-      group?.issueCategory === IssueCategory.PERFORMANCE &&
-      event.type === 'transaction'
-    ) {
-      return [];
-    }
-
-    let path = `/projects/${organization.slug}/${projectSlug}/events/${event.id}/grouping-info/`;
-    if (this.state?.configOverride) {
-      path = `${path}?config=${this.state.configOverride}`;
-    }
-
-    return [['groupInfo', path]];
-  }
-
-  getDefaultState() {
-    return {
-      ...super.getDefaultState(),
-      isOpen: false,
-      configOverride: null,
-    };
-  }
-
-  toggle = () => {
-    this.setState(state => ({
-      isOpen: !state.isOpen,
-      configOverride: state.isOpen ? null : state.configOverride,
-    }));
-  };
-
-  handleConfigSelect = selection => {
-    this.setState({configOverride: selection.value}, () => this.reloadData());
-  };
-
-  generatePerformanceGroupInfo() {
-    const {group, event} = this.props;
-    const {occurrence} = event;
-    const {evidenceData} = occurrence as EventOccurrence;
-
-    const variant = group
-      ? {
-          [group.issueType]: {
-            description: t('performance problem'),
-            hash: occurrence?.fingerprint[0] || '',
-            hasMismatch: false,
-            key: group.issueType,
-            type: 'performance-problem',
-            evidence: {
-              op: evidenceData?.op,
-              parent_span_ids: evidenceData?.parentSpanIds,
-              cause_span_ids: evidenceData?.causeSpanIds,
-              offender_span_ids: evidenceData?.offenderSpanIds,
-            },
+function generatePerformanceGroupInfo({event, group}: {event: Event; group: Group}) {
+  const {occurrence} = event;
+  const {evidenceData} = occurrence as EventOccurrence;
```

Comment:
Why don't we just fix the type of `event` to omit this force type cast? Force type casting will return in false positives in type checking.

Reply:
Can do, was being lazy and just copy-pasting what was there before